src/gc/env/gcenv.safeint.h

```diff
@@ -0,0 +1,810 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
@jkotas We also use S_SIZE_T for init() (see gc.cpp:5059), although looking at the code I don't think it is currently protecting anything. It is written as `S_SIZE_T safe_sniff_buf_size = S_SIZE_T(1 + n_heaps*n_sniff_buffers + 1);`, which seems like any overflow would happen in the calculation before it is passed to the S_SIZE_T constructor.

I'm fine with getting rid of the file, I agree it's pretty heavyweight for how it's used. I was just being cautious because it's tricky to write overflow code so the compiler doesn't defeat it